homeassistant/components/sensoterra/coordinator.py

```diff
@@ -0,0 +1,68 @@
+"""Polling coordinator for the Sensoterra integration."""
+
+from collections.abc import Callable
+from datetime import timedelta
+
+from sensoterra.customerapi import (
+    CustomerApi,
+    InvalidAuth as ApiAuthError,
+    Timeout as ApiTimeout,
+)
+from sensoterra.probe import Sensor
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import LOGGER, SCAN_INTERVAL_MINUTES
+from .models import SensoterraSensor
+
+
+class SensoterraCoordinator(DataUpdateCoordinator[list[SensoterraSensor]]):
+    """Sensoterra coordinator."""
+
+    def __init__(self, hass: HomeAssistant, api: CustomerApi) -> None:
+        """Initialize Sensoterra coordinator."""
+        super().__init__(
+            hass,
+            LOGGER,
+            name="Sensoterra probe",
+            update_interval=timedelta(minutes=SCAN_INTERVAL_MINUTES),
+        )
+        self.api = api
+        self.add_devices_callback: Callable[[list[SensoterraSensor]], None] | None = (
+            None
+        )
```

Comment:
You can just use the normal listener setup from the coordinator

Reply:
I'm a bit lost here, could you please clarify a bit? 

The idea is that the API gives a list of probes with sensors, and that we might need to add newly registered sensors to HA.